### PR TITLE
Events: Skippable Item/Feat/Party/Skill Events

### PR DESCRIPTION
**Pull request by Daztek**
- The OnUseItem and OnUseFeat events are now skippable.
- Exposed the following event data for OnUseItem:
ITEM_SUB_PROPERTY_INDEX int 
TARGET_POSITION_X float       
TARGET_POSITION_Y float       
TARGET_POSITION_Z float      
- Added skippable Party events
- Added skippable Skill events

### CODE REVIEW

File: Plugins/Events/Events/FeatEvents.cpp [hunk 1]
```diff
@@ -49,7 +69,9 @@ void FeatEvents::UseFeatHook
     Events::PushEventData("TARGET_POSITION_Y", pvTarget ? std::to_string(pvTarget->y) : "0.0");
     Events::PushEventData("TARGET_POSITION_Z", pvTarget ? std::to_string(pvTarget->z) : "0.0");
 
-    Events::SignalEvent(before ? "NWNX_ON_USE_FEAT_BEFORE" : "NWNX_ON_USE_FEAT_AFTER", thisPtr->m_idSelf);
+    Events::SignalEvent("NWNX_ON_USE_FEAT_AFTER", thisPtr->m_idSelf);
```
Comment (mtijanic) [inline Plugins/Events/Events/FeatEvents.cpp]: Do we want to signal the AFTER event if it was bypassed?

Comment (Daztek) [reply Plugins/Events/Events/FeatEvents.cpp]: It's signaled in BhaalM's Healer's Kit event and Liareth's [example](https://github.com/Liareth/unified/blob/ff715cc2acd889f21aaf86f6426393d6f98ef66e/Plugins/Events/Events/AssociateEvents.cpp) code from #16, so probably? :D

**Comment by Daztek** [comment]
Now with 100% more CombatMode events

**Comment by Daztek** [comment]
- Removed CombatModeEvents since it's handled by NWNX_CombatModes now
- Party events are now skippable

**Comment by Daztek** [comment]
- Added skippable Skill events
This implements #247 and partially implements #34

File: Plugins/Events/Events/SkillEvents.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    uint8_t subSkill,
+    API::Types::ObjectID target,
+    NWNXLib::API::Vector targetPosition,
+    API::Types::ObjectID area,
+    API::Types::ObjectID usedItem,
+    int32_t activePropertyIndex)
+{
+    int32_t retVal;
+
+    Events::PushEventData("SKILL_ID", std::to_string(skill));
```
Comment (mtijanic) [inline Plugins/Events/Events/SkillEvents.cpp]: I actually like the lambda approach as opposed to duplicated code. When you push all this stuff, you can wrap it as:
```C
auto Push = [&]() { /* all the push events here */ };
```
Then you just call it as `Push()` both before and after the event. Removes code duplication.

Can also make it look like:
```C
auto PushAndSignal = [&](std::string &ev) -> bool {
    Events::PushEventData("...", ...);
    ...
    return EventsSignal(ev, thisPtr->m_idSelf);
```

And then your core function is:

```C
if (PushAndSignal("NWNX_ON_USE_SKILL_BEFORE"))
{
    retVal = ...;
}

PushAndSignal("NWNX_ON_USE_SKILL_AFTER");
return retVal;
```

And same logic can be applied to anything that needs to push more than 1-2 pieces of event data.

Comment (Daztek) [reply Plugins/Events/Events/SkillEvents.cpp]: Added, that does make it look a lot more readable

**Comment by mtijanic** [comment]
I think this is good to merge now, but let's wait for @Liareth to take a look as well.

**Pull request merged**